lib/server.js

```diff
@@ -175,7 +183,8 @@ function handleRequest (opts, req, resp) {
             query: urlData.query,
             method: req.method.toLowerCase(),
             headers: req.headers,
-            body: body
+            body: body,
+            reqId: reqId
```

Comment:
This seems to be the cleanest way because then `headers['x-request-id']` is not carried over to all child requests. Instead, it's only set in `RESTBase.reqId`, so it *should* be more performant like this.